### ISSUE TITLE
Alga traits not implemented by statically-sized matrices/vectors?

### ISSUE THREAD

**Issue by YPares** [dimforge/nalgebra#1311]
Is there a reason why https://github.com/dimforge/nalgebra/blob/dev/src/third_party/alga/alga_matrix.rs implements the alga::linear traits only for the OMatrix variant? (and not e.g. for Matrix in general or separately for SMatrix or SVector?)

**Comment by Ralith** [comment]
Isn't `OMatrix` a superset of `SMatrix`?

**Comment by YPares** [comment]
@Ralith Really? Looking at the types I doesn't seem clear if that's the case. Their allocator types seem entirely different.

**Comment by YPares** [comment]
After testing it, you are right. (But I still don't understand how rustc may consider the two types to be compatible, though)

Though DVector does not implement those alga::linear traits (this time it doesn't work), is there a reason? I originally wanted to use alga::linear traits to abstract over statically-sized and dynamically sized vectors.

**Comment by YPares** [comment]
Yeah, I did it this way originally but using alga traits led to much clearer type signatures. That's also the way I like to develop: the less my code knows about the underlying structure, the less likely I am to introduce bugs by accidentally relying on some feature that I shouldn't have relied on. (For instance, when switching from SVector to T: VectorSpace, I realized I had inadvertently been using Copy in some place which didn't need it)
Also, the code I'm writing is meant to back up a research paper, and that way the less implementation "noise" there is, the better the code reflects the main idea of the paper.
